fili-core/src/main/java/com/yahoo/bard/webservice/data/config/provider/MakerDictionary.java

```diff
@@ -0,0 +1,295 @@
+                            LongMaxMaker.class,
+                            LongMinMaker.class,
+                            LongSumMaker.class,
+                            RowNumMaker.class
+
+                            // These (and a few others) are built in but require some arguments
+                            // So, you'll have to list them as custom makers with their parameters
+                            // AggregationAverageMaker.class,
+                            // CardinalityMaker.class,
+                    )
+            )
+    );
+
+    // Cache of default metric makers
+    private static MakerDictionary defaultMakers;
+
+    /**
+     * Get a map of name -&gt; makers for the default (built-in) MetricMakers.
+     *
+     * Makers included by default:
+     *
+     * 1. Are in the same package as MetricMaker
+     * 2. Have a constructor that takes a single MetricDictionary
+     *
+     * @param dict metric dictionary to use for built-in makers
+     * @return the dictionary of MetricMakers
+     */
+    public static MakerDictionary getDefaultMakers(MetricDictionary dict) {
+        if (defaultMakers == null) {
+            defaultMakers = new MakerDictionary();
+            for (Class<? extends MetricMaker> cls : BUILTIN_METRIC_MAKERS) {
+                // LongSumMaker => longSum
+                String makerName = lowerFirst(cls.getSimpleName()).replaceFirst("Maker", "");
+                MetricMaker actualMaker = instantiateObjectFromArgs(cls, dict);
+                if (actualMaker != null) {
+                    defaultMakers.put(makerName, actualMaker);
+                } else {
+                    throw new ConfigurationError("Unable to instantiate built-in metric maker " + cls);
+                }
+            }
+        }
+
+        return defaultMakers;
+    }
+
+    /**
+     * Initialize the metric makers, custom and built-in.
+     *
+     * @param dict MetricDictionary to use for every maker
+     * @param mDict Dictionary of function name to MetricMaker
+     * @param makers user-defined config listing custom MetricMakers
+     */
+    public static void loadMetricMakers(
+            MetricDictionary dict,
+            MakerDictionary mDict,
+            Map<String, MakerConfiguration> makers
+    ) {
+        loadMetricMakers(dict, mDict, makers, false);
+    }
+
+    /**
+     * Initialize the metric makers, custom and optionally built-in.
+     *
+     * Eventually, 'skipDefault' will be an option in the YAML config.
+     *
+     * @param dict MetricDictionary to use for every maker
+     * @param mDict Dictionary of function name to MetricMaker
+     * @param makers user-defined config listing custom MetricMakers
+     * @param skipDefault true to skip loading built-in metrics
+     */
+    public static void loadMetricMakers(
+            MetricDictionary dict,
+            MakerDictionary mDict,
+            Map<String, MakerConfiguration> makers,
+            boolean skipDefault
```

Comment:
Is there a reason you're not using your custom class 'ConfigurationDictionary<T>' here instead of crafting a raw map?